collect_app/src/main/java/org/odk/collect/android/widgets/ExDecimalWidget.java

```diff
@@ -68,6 +68,5 @@ protected void fireActivity(Intent i) throws ActivityNotFoundException {
     public void setBinaryData(Object answer) {
         DecimalData decimalData = ExternalAppsUtils.asDecimalData(answer);
         answerText.setText(decimalData == null ? null : decimalData.getValue().toString());
-        widgetValueChanged();
```

Comment:
This is because the call on `setText` triggers the textchanged listener, right? Good catch.